packages/web-components/case-card/src/element.ts

```diff
@@ -0,0 +1,143 @@
+// import css from '@gemeente-denhaag/card/src/css.mjs';
+import css from './css.js';
+
+export interface CardData {
+  dateTime: string;
+  href: string;
+  heading: string;
+}
+// const html = (str: string) => str;
+
+export const QUOTATION_MARK = 0x0022;
+export const AMPERSAND = 0x0026;
+export const LESS_THAN_SIGN = 0x003c;
+export const GREATER_THAN_SIGN = 0x003e;
+export const SPACE = 0x0020;
+export const NONCHARACTER_FFFE = 0xfffe;
+export const NONCHARACTER_FFFF = 0xffff;
+
+/**
+ * Escaping of the following characters using entities: & < > "
+ * Also replace non-character ranges with the replacement character (U+FFFD)
+ *
+ * @param {string} xml
+ * @param {boolean=} attr When false, double quotes will not be escaped
+ * @return {string}
+ */
+export const escapeXML = function escapeXML(xml: string) {
+  /**
+   * Escape the Unicode 'non-characters'
+   * Escape characters below 0xC0 except tab and the newlines
+   * Escape < to prevent starting an element
+   * Escape > to prevent ending a CDATA section
+   * Escape " to prevent ending an attribute
+   * Escape & to prevent starting an entity reference
+   * Don't escape surrogate blocks, because they are essential for Unicode support in JavaScript
+   *
+   * @see http://www.w3.org/TR/REC-xml/#charsets
+   * @param {string} m One character.
+   * @return {string} The same character, or its escaped version.
+   */
+  function escapeChar(char: string) {
+    let m = char;
+    const charCode = m.charCodeAt(0);
+
+    if (charCode === QUOTATION_MARK) {
+      // && attr !== false)
+      // One byte shorter than than &quot;
+      m = '&#34;';
+    } else if (charCode === AMPERSAND) {
+      // &
+      m = '&amp;';
+    } else if (charCode === GREATER_THAN_SIGN) {
+      /*
+       * In a text node the > needn't be escaped,
+       * but in CDATA sections it must be.
+       */
+      // <
+      m = '&gt;';
+    } else if (charCode === LESS_THAN_SIGN) {
+      m = '&lt;';
+    } else if (charCode < SPACE || charCode === NONCHARACTER_FFFE || charCode === NONCHARACTER_FFFF) {
+      m = '\uFFFD';
+    }
+
+    return m;
+  }
+
+  // TODO: Move RegExp to constant
+
+  // eslint-disable-next-line no-control-regex
+  return xml.replace(/[\0-\t\u000B\u000C\u000E-\u00C0"&<>\uFFFE\uFFFF]/g, escapeChar);
+};
+
+export class DenhaagCaseCardElement extends HTMLElement implements CardData {
+  dateTime: string;
+  href: string;
+  heading: string;
+  _shadow: ShadowRoot;
+  _div: HTMLElement;
+  constructor() {
+    super();
+    this._shadow = this.attachShadow({
+      mode: 'closed',
+    });
+    const style = this.ownerDocument.createElement('style');
+    style.textContent = css;
+    this._shadow.appendChild(style);
```

Comment:
Ik denk dat we even in de NL Portal moeten testen hoe we dit werkend kunnen krijgen met een CSP header zonder unsafe-inline, aangezien dat vanuit DigiD vereist is. Ik gok dat we hier ook een optie moeten toevoegen om een nonce attribuut toe te voegen.

Reply:
Heel goed punt van die unsafe inline, we moeten het inderdaad even op een andere manier aanpakken met `createElement` ipv `innerHTML`.

Ja, laten we dat testen! We willen voor de Fieldlab van juni deze component testen met >= 2 thema's, in Open Web Concept en in een andere omgeving 😀

Maar eerder mag altijd 😉

Reply:
De styles gebeuren nu niet meer met een inline style element maar met constructed stylesheets. Zo komt het component tot nu toe door de CSP bij mijn lokale test.